fusion-plugin-react-router/__tests__/plugin.js

```diff
@@ -412,6 +414,51 @@ if (__BROWSER__) {
     const simulator = setup(app);
     await simulator.render('/');
   });
+
+  test('Browser History Provider', async () => {
+    const element = <div />;
+
+    const app = getApp(element);
+    const UniversalEvents = getMockEvents({
+      title: '/',
+      page: '/',
+    });
+
+    let middlewareHistory;
+    let providerHistory;
+
+    app.register(UniversalEventsToken, UniversalEvents);
+    app.register(BrowserHistoryToken, browserHistoryPlugin);
+    app.register(
+      createPlugin({
+        deps: {
+          history: BrowserHistoryToken,
+        },
+        provides({history}) {
+          expect(history).toBeTruthy();
+          providerHistory = history;
+        },
+      })
+    );
+
+    app.middleware(
+      {
+        router: RouterToken,
+      },
+      ({router}) => (ctx, next) => {
+        const {history} = router.from(ctx);
+        expect(history).toBeTruthy();
+        middlewareHistory = history;
+        return next();
+      }
+    );
+    const simulator = setup(app);
+    await simulator.render('/');
+
+    expect(middlewareHistory).toBe(providerHistory);
```

Comment:
I mostly just want to assert that the instance is the same regardless of where you get it.

Is there anything else I should add to the tests?

Reply:
this is good